packages/core/src/components/forms/_input.scss

```diff
@@ -29,6 +29,35 @@ Styleguide input
     @include pt-input-large();
   }
 
+  &.#{$ns}-small {
+    .#{$ns}-button {
+      @include pt-button-height($pt-button-height-smaller);
+      margin: ($pt-input-height-small - $pt-button-height-smaller) / 2;
+    }
+
+    .#{$ns}-tag {
+      @include pt-button-height($pt-button-height-smaller);
+      margin: ($pt-input-height-small - $pt-button-height-smaller) / 2;
+    }
+
+    > .#{$ns}-icon,
+    .#{$ns}-input-action > .#{$ns}-spinner {
+      margin: ($pt-input-height-small - $pt-icon-size-standard) / 2;
+    }
+
+    .#{$ns}-input {
```

Comment:
these styles should be at the top level, because the selector on line 32 becomes `.ns-input.ns-small`. remove this nesting.

Reply:
![image](https://user-images.githubusercontent.com/464822/45882219-59e16f80-bda5-11e8-8937-d6b3f9aa9028.png)

modifier has no effect.